third_party/terraform/utils/config_test.go

```diff
@@ -90,6 +90,34 @@ func TestAccConfigLoadValidate_credentials(t *testing.T) {
 	}
 }
 
+func TestAccConfigLoadValidate_impersonated(t *testing.T) {
+	if os.Getenv(TestEnvVar) == "" {
+		t.Skip(fmt.Sprintf("Network access not allowed; use %s=1 to enable", TestEnvVar))
+	}
+	testAccPreCheck(t)
+
+	serviceaccount := multiEnvSearch([]string{"GOOGLE_IMPERSONATE_SERVICE_ACCOUNT"})
+	proj := getTestProjectFromEnv()
+
+	config := &Config{
```

Comment:
Could we add `Credentials` to this config? And if we end up implementing a similar block in the `AccessToken` block, it might not be bad to write a test for that as well. Also, do we want to try and write a test with delegates?

Reply:
Fixed